docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Custom fields reference - Strapi Developer Docs
+description: Learn how you can use custom fields to extend Strapi's content-types capabilities.
+sidebarDepth: 3
+canonicalUrl: https://docs.strapi.io/developer-docs/latest/development/custom-fields.html
+---
+
+# Custom fields
+
+Custom fields extend Strapi’s capabilities by adding new types of fields to content-types and components. Once created or installed, custom fields can be used in the Content-Types Builder and Content Manager just like built-in fields.
+
+The present documentation is intended for custom field creators: it describes which APIs and functions developers must use to create a new custom field. The [user guide](#/user-docs/latest/plugins/introduction-to-plugins.md#custom-fields) describes how to install and use custom fields from Strapi's admin panel.
+
+<!-- TODO: uncomment and adjust content when blog post is ready -->
+<!-- ::: strapi Prefer to learn by building?
+If you'd rather directly jump to a concrete example, see our [Creating a color custom field guide](#) page for step-by-step instructions on how to build your first custom field from scratch.
+::: -->
+
+It is recommended that you develop a dedicated [plugin](/developer-docs/latest/development/plugins-development.md) for custom fields. Custom-field plugins include both a server and admin panel part. The custom field must be registered in both parts before it is usable in Strapi's admin panel.
+
+Once created and used, custom fields are defined like any other attribute in the model's `schema.json`. An attribute using a custom field will have its type represented as `customField` (i.e. `type: 'customField'`). Depending on the custom field being used a few additional properties may be present in the attribute's definition (see [models documentation](/developer-docs/latest/development/backend-customization/models.md#custom-fields)).
```

Comment:
We don't need to explicitly define the `schema.json` here as this isn't exactly required, you can also use it in schema.js for example which many plugins use. This is probably a nit-pick so up to you, just figured I'd mention it.

We could generalize this by just saying "schema" instead of specifying the filetype

Reply:
Great idea, thanks!